### ISSUE TITLE
LB instance targets are not reachable for sometime after they are restarted.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#6647]
Rancher Version: Build from master
rancher/dns:v0.10.5
rancher/healthcheck:v0.1.0
rancher/net:v0.7.0
rancher/metadata:v0.6.3
rancher/plugin-manager:v0.2.1
rancher/scheduler:v0.1.2

Steps to reproduce the problem:
Create a LB service with target services.
Attempt any of the following actions:
1. Restart 1 instance of the target services.
2. Stop  1 instance of the target services and wait for it to start.
3.  Deactivate/Activate environment. 

Access LB. First set of hits to LB ip , does not redirect traffic to the restarted instance.

```
admin_client = <cattle.Client object at 0x7f169a661d10>
client = <cattle.Client object at 0x7f169a7a81d0>, socat_containers = None

    def test_lb_services_restart_instance(
            admin_client, client, socat_containers):
    
        port = "19029"
    
        service_scale = 3
        lb_scale = 1
    
        env, service, lb_service = create_environment_with_balancer_services(
            admin_client, client, service_scale, lb_scale, port)
    
        validate_lb_service(admin_client, client, lb_service, port, [service])
    
        # restart instance
        container_name = \
            env.name + FIELD_SEPARATOR + service.name + FIELD_SEPARATOR + "1"
        containers = client.list_container(name=container_name)
        assert len(containers) == 1
        container = containers[0]
        container = client.wait_success(container.restart(), 120)
        wait_for_condition(client, container,
                           lambda x: x.state == 'running',
                           lambda x: 'State is: ' + x.state)
        wait_for_condition(client, container,
                           lambda x: x.startCount == 2,
                           lambda x: 'State is: ' + x.state)
        time.sleep(30)
        wait_for_lb_service_to_become_active(admin_client, client,
                                             [service], lb_service)
>       validate_lb_service(admin_client, client, lb_service, port, [service])

tests/v2_validation/cattlevalidationtest/core/test_services_lb_balancer.py:416: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:1000: in validate_lb_service
    test_ssl_client_con)
tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:1029: in validate_lb_service_con_names
    check_round_robin_access(container_names, host, port)
tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:1900: in check_round_robin_access
    hostheader=hostheader, path=path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

container_names = ['526a456245a1', '4cf16129a5be', '09c0f3586d38']
lb_ip = '104.154.138.104', port = '19029', hostheader = None
path = '/name.html'

    def check_round_robin_access_lb_ip(container_names, lb_ip, port,
                                       hostheader=None, path="/name.html"):
    
        con_hostname = container_names[:]
        con_hostname_ordered = []
    
        url = "http://" + lb_ip +\
              ":" + port + path
    
        logger.info(url)
    
        headers = None
        if hostheader is not None:
            headers = {"host": hostheader}
    
        logger.info(headers)
    
        for n in range(0, len(con_hostname)):
            if headers is not None:
                r = requests.get(url, headers=headers)
            else:
                r = requests.get(url)
            response = r.text.strip("\n")
            logger.info(response)
            r.close()
>           assert response in con_hostname
E           assert '09c0f3586d38' in ['526a456245a1']

tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:1928: AssertionError
```

Following are the 3 tests that fail:

cattlevalidationtest.core.test_services_lb_balancer.test_lb_services_restart_instance
cattlevalidationtest.core.test_services_lb_balancer.test_lb_services_stop_start_instance
cattlevalidationtest.core.test_services_lb_balancer.test_lb_services_deactivate_activate_environment

**Comment by alena1108** [comment]
I've reproduced the delay (although it wasn't 30 seconds). Haproxy is configured correctly with the backend ip address, but the backend container is not accessible via ip sec network for a while. It's up and running (I've checked docker status), it is just not reachable by its ip address.

@leodotcloud the way to reproduce it is - you don't even need an lb. Just start any client service with "curl" installed, then create nginx service. Test nginx service by:

```
curl 10.42.9.92:80
```

from the client container. Restart nginx container, continue curling the nginx ip from curl. Even when nginx container is up and running (which happens immediately), the ip address is not accessible for a while.

Once debugged, see if this slowness is acceptable. If yes, let @sangeethah know if longer timeouts have to be configured in validation tests

**Comment by leodotcloud** [comment]
@sangeethah Is this issue still present?

**Comment by sangeethah** [comment]
@leodotcloud , yes this issue is still reproducible in automation runs.
After a instance is restarted , LB will not direct traffic to the restarted instance for the first few hits of LB ip.

**Comment by sangeethah** [comment]
Tested with rancher-server -v1.4.0-rc5.

Following automation tests relating to Lb instance restart does not report this issue anymore:

1.Restart 1 instance of the target services.
2.Stop 1 instance of the target services and wait for it to start.
3.Deactivate/Activate environment.

**Comment by sangeethah** [comment]
@leodotcloud , when testing with rancher-server -v1.4.1-rc1 , I see this issue starting to happen again.

**Comment by leodotcloud** [comment]
@sangeethah As discussed offline, in your setup the ipsec version is `v0.8.1`. The fix went in `v0.8.3`. The catalog is pointing to the wrong tag.
